lib/easy.c

```diff
@@ -771,6 +793,9 @@ static CURLcode dupset(struct Curl_easy *dst, struct Curl_easy *src)
   /* duplicate all strings */
   for(i = (enum dupstring)0; i< STRING_LASTZEROTERMINATED; i++) {
     result = Curl_setstropt(&dst->set.str[i], src->set.str[i]);
+    /* Curl_setstropt return CURLE_BAD_FUNCTION_ARGUMENT with blob */
+    if(result == CURLE_BAD_FUNCTION_ARGUMENT)
```

Comment:
I don't think we should assume that this return code will always be benign and mean its a blob. I think the blob should be better special cased than pretending its a string.